Fix issues preventing library from building with Qt 6

Cuperino opened the pull request:
Changes:
1. CMake instructions now handle both Qt 6 and Qt 5
2. Pass correct QString-like type to setWindowTitle to fix build error
3. Remove qAsConst optimization to address deprecation warning
4. Increase CMake minimum version to address old CMake versions deprecation warning
5. Perform the install step after the Framework target property is set and add Framework indicator to the install call to address warning regarding possible incorrect path warning
6. Add CMakeLists.txt.user and build directories to .gitignore
7. Update GitHub workflow by adding Qt 6 and bumping macOS version

Cuperino committed:
Fix library potentially being stored in non-framework path

1. Perform the install step after the Framework target property is set
2. Add Framework indicator to the install call

Cuperino committed:
Add CMakeLists.txt.user and build directories to .gitignore

dantti commented on .github/workflows/build.yml:
```diff
@@ -13,14 +13,16 @@ jobs:
           - Release
         qt:
-          - version: "5.15.2"
+          - version: "5.15.14"
```
This release doe not exist as Open Source, it will just fallback to 15.2

Cuperino replied:
I saw that it was added to the opensource archive in late May (just a few days ago): https://download.qt.io/archive/qt/5.15/

dantti replied:
look at the build logs, aqtinstall does not know it

dantti replied:
also that link only provides sources, QtC did not release compiled versions of it

dantti replied:
It exists as open source bacause of their agreement but only as source, as binaries it's only available for comercial users.

Cuperino replied:
Well, that explains why such a seemingly recent build wouldn't work with MacOS 14...

dantti commented:
Also while you are at it, use macos-latest (& friends), the create build dir can be removed, and for Qt5.15 macos you can fix it by forcing only x86_64 builds, I fixed a few other kdab repos due new macos-latest being AppleSilicon now

dantti commented:
see https://github.com/KDAB/GammaRay/blob/master/.github/workflows/build.yml  you need to pass it to cmake

dantti commented on .github/workflows/build.yml:
```diff
@@ -13,14 +13,17 @@ jobs:
+            architecture: "x86_64"
+          - version: "6.4.*"
+            requested: "6.4"
+            architecture: "arm64"
```
now that you did this, I'm actually thinking I should actually have done this to GammaRay too, but actually it would be nicer if it was "x86_64;arm64" as that would build both, maybe you should try this here too.

Cuperino replied:
For Qt 6, good idea.

Cuperino replied:
`x86_64;arm64` didn't work, I've worked around it by making adding another Qt 6 entry, one for each architecture.
```
          - version: "5.15.2"
            requested: "5.15"
            architectures: "x86_64"
          - version: "6.4.*"
            requested: "6.4"
            architectures: "x86_64"
          - version: "6.4.*"
            requested: "6.4"
            architectures: "arm64"
```

dantti replied:
Maybe you forgot the quotes? because `;` char can finish the cmd line

Cuperino replied:
Commit 8d935b3d018daec4de858fa8add32c65bf97a8ba looked like this:
```
            architectures: "x86_64;arm64"
```
```
          -DCMAKE_OSX_ARCHITECTURES=${{ matrix.qt.architectures }}
```

dantti replied:
would this work? -DCMAKE_OSX_ARCHITECTURES="${{ matrix.qt.architectures }}"

Cuperino replied:
That worked! Thanks

dantti commented on CMakeLists.txt:
```diff
@@ -19,10 +19,10 @@ set(${PROJECT_NAME}_VERSION ${${PROJECT_NAME}_VERSION_MAJOR}.${${PROJECT_NAME}_V
 if(CMAKE_VERSION VERSION_LESS "3.1")
     if(CMAKE_COMPILER_IS_GNUCXX)
-        set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+        set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++17")
```
I think you should remove this IF entirely since the minimum version you set to 3.12 already

Cuperino replied:
CMake's bump was related to a CMake warning, so I should probably remove this.  Now, on the topic of older versions of C++, since I didn't end up making use of `std::as_const` as I originally intended, we could continue to leave the minimum at C++11...

@dantti If I were to leave the minimum at 11, would we need to use a lower CMake version as well?

dantti replied:
I don't think so 3.12 is old enough to raise it I would say, and we are likely the only users of this library...

Cuperino replied:
I agree, 3.12 is old enough. I don't see a need for supporting C++ 11 but since we also don't need to bump it, we can leave it as it is until the need comes up.

Cuperino committed:
Fix issues preventing library from building with Qt 6

1. CMake instructions now handle both Qt 6 and Qt 5
2. Pass correct QString-like type to setWindowTitle
3. Remove qAsConst optimization to address deprecation warning

Cuperino committed:
Bump version number since these changes finish adding Qt 6 support

Cuperino committed:
Increase CMake minimum version

This addresses CMake versions < 3 deprecation warning

dantti reviewed (approved):
LGTM
